draft-trammell-taps-post-sockets.md

```diff
@@ -701,23 +718,88 @@ func receiveMulticast() {
 }
 ~~~~~~~~
 
+## Association Bootstrapping
+
+Here, we show how Association state may be initialized. The goal is to
+create a long-term Association from which Carriers may be derived and,
+if possible, used immediately. We will describe two variations -- one
+where the Association is configured with trust model information
+necessary for the Remote to establish a connection, and another
+where the Association is created from a pre-shared key established
+out-of-band mechanism.
+
+### Association Preparation
+
+Perhaps the more common case for creating Associations is to initialize
+new connections to Remotes. To do so, Associations must possess the
+security information needed to create a fresh connection.
+Per {{I-D.ietf-taps-transport-security}}, this includes, at a minimum:
+
+- An identity and access or an interface to the associated private key.
+- Trust model constraints, such as pinned public keys and anchor certificates.
+
+The following example shows how an Association can be created with this
+information to be later used for creating a remote connection.
+
+~~~~~~~~
+// create an association with a given identity and set of trusted certificates
+func createAssociation(ident Identity, trustedCerts []Certificate) Association {
+    association := Association()
```

Comment:
Mmm I think the association always needs a remote, based on its definition. (I think there can be room for a parent object or parameters that allow it to be easy to set the same local identity and trusted config for multiple associations, but I agree that for this definition you need a single remote on every association)